dp_check/dp_check.go

```diff
@@ -142,31 +233,101 @@ func main() {
 	if syscall.Getuid() != 0 {
 		infoLog.Println("Not running as root, some checks may fail.")
 	}
-	// Check IPv6
-	runCheck("IPv6 routes", func() error {
-		var out string
-		var err error
-		cmd := "ip -6 route show"
-		infoLog.Printf("Check IPv6 routes with:|%v|...\n", cmd)
-		if out, err = getOutput(cmd); err != nil {
+
+	// Check if dp_check is running on GCP
+	runCheck("Running on GCP", func() error {
+		ret, err := isRunningOnGCP()
+		if err != nil {
 			return err
 		}
-		if !strings.Contains(out, "2001:4860:8040::/42") {
-			return fmt.Errorf(`Missing route prefix to backends: 2001:4860:8040::/42.
-IPv6 route setup either failed or hasn't been attempted`)
+		if !ret {
+			return fmt.Errorf("dp_check is not running on GCP, this tool will not work as intended")
 		}
 		return nil
 	})
+
+	// Check connection to metadata server
+	runCheck("DirectPath enablement", func() error {
+		client := &http.Client{}
+		metadataServerUrl := "http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ipv6s"
+		infoLog.Printf("Check if this VM enables DirectPath by sending http GET request to %s", metadataServerUrl)
+		req, err := http.NewRequest("GET", metadataServerUrl, nil)
+		if err != nil {
+			return err
+		}
+		req.Header.Add("Metadata-Flavor", "Google")
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		ipv6FromMetadataServer = net.ParseIP(strings.TrimSuffix(string(body), "\n"))
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode == 200 {
+			return nil
```

Comment:
in this 200 response code case, can we log the IPv6 address that we expect to find on a local interface?